CMakeLists.txt

```diff
@@ -66,7 +75,7 @@ IF ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
    MESSAGE("")
    MESSAGE("cmake for GCC ")
    SET(PLATFORM_LINK_LIBRIES rt)
-   SET(CMAKE_CXX_FLAGS "-Wall -rdynamic -Wunused -std=c++14 -pthread -D_GLIBCXX_USE_NANOSLEEP")
+   SET(CMAKE_CXX_FLAGS "-fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free -Wall -rdynamic -Wunused -std=c++14 -pthread -D_GLIBCXX_USE_NANOSLEEP")
```

Comment:
@john-gress I did not remove tcmalloc. I added the missing flags for the open source community users. These flags are the ones that gperftools (tcmalloc) recommends the tcmalloc using library to have. 

From our spec file I do not see them and that makes me wonder that we might not be using TCMALLOC correctly in our current source code.